syntax = "proto3";
service CatalogService {
    rpc query(QueryRequest) returns (QueryResponse){}
    rpc buy(BuyRequest) returns(BuyResponse){}
}
message QueryRequest {
    string name = 1;
}

message BuyRequest {
    string name = 1;
    int32 quantity = 2;
}


message QueryResponse {
    int32 quantity = 1;
    float price = 2;
}

message BuyResponse {
    int32 done = 1;
}